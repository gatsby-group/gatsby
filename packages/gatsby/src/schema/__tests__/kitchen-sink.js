// @flow

// Children
// childX
// childrenX
// NODE__ connections
// addTypeDefs and corner cases
// addResolvers

const fs = require(`fs`)
const path = require(`path`)

const { graphql } = require(`graphql`)
const { store } = require(`../../redux`)
const { build } = require(`../index`)
const withResolverContext = require(`../context`)

jest.mock(`../../utils/api-runner-node`)
const apiRunnerNode = require(`../../utils/api-runner-node`)

// XXX(freiksenet): Expand
describe(`Kichen sink schema test`, () => {
  let schema

  const runQuery = query =>
    graphql(schema, query, undefined, withResolverContext({}, schema))

  beforeAll(async () => {
    apiRunnerNode.mockImplementation((api, ...args) => {
      if (api === `setFieldsOnGraphQLNodeType`) {
        return mockSetFieldsOnGraphQLNodeType(...args)
      } else if (api === `addResolvers`) {
        return mockAddResolvers(...args)
      } else {
        return []
      }
    })
    const nodes = JSON.parse(
      fs.readFileSync(path.join(__dirname, `./fixtures/kitchen-sink.json`))
    )
    store.dispatch({ type: `DELETE_CACHE` })
    nodes.forEach(node =>
      store.dispatch({ type: `CREATE_NODE`, payload: node })
    )
    await build({})
    schema = store.getState().schema
  })

  it(`passes kitchen sink query`, async () => {
    expect(
      await runQuery(`
        {
          allPostsJson(sort: { fields: likes, order:DESC}) {
            edges {
              node {
                id
                time
                code
                likes
                comment
                image {
                  childImageSharp {
                    id

        					}
                }
        			}
            }
          }
        }
    `)
    ).toMatchSnapshot()
  })
})

const mockSetFieldsOnGraphQLNodeType = ({ type: { name } }) => []

const mockAddResolvers = ({ addResolvers }) => {}
