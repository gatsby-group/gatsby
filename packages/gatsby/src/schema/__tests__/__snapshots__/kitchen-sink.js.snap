// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Kichen sink schema test passes kitchen sink query 1`] = `
Object {
  "data": Object {
    "addResolvers": Array [
      Object {
        "code": "BdiU-TTFP4h",
        "id": "1685001452849004065",
        "likes": 33,
      },
      Object {
        "code": "BdIpEzalAJx",
        "id": "1677771511187112561",
        "likes": 29,
      },
    ],
    "filter": Object {
      "edges": Array [
        Object {
          "node": Object {
            "comment": 1,
            "id": "1270677182602272387",
          },
        },
        Object {
          "node": Object {
            "comment": 0,
            "id": "1256134251849702933",
          },
        },
      ],
    },
    "resolveFilter": Object {
      "id": "1601601194425654597",
      "idWithDecoration": "decoration-1601601194425654597",
      "likes": 9,
    },
    "sort": Object {
      "edges": Array [
        Object {
          "node": Object {
            "_3invalidKey": null,
            "code": "BGiWipNM96D",
            "comment": 1,
            "id": "1270677182602272387",
            "idWithDecoration": "decoration-1270677182602272387",
            "image": Object {
              "childImageSharp": Object {
                "id": "081cbcc5-44ec-5f4e-ae7a-fc25e68ff615",
              },
            },
            "likes": null,
            "time": "12.06.2016",
          },
        },
        Object {
          "node": Object {
            "_3invalidKey": null,
            "code": "BFur3Jfs94V",
            "comment": 0,
            "id": "1256134251849702933",
            "idWithDecoration": "decoration-1256134251849702933",
            "image": Object {
              "childImageSharp": Object {
                "id": "a0a75092-ed13-5117-a44b-7360940f99b5",
              },
            },
            "likes": null,
            "time": "23.05.2016",
          },
        },
      ],
    },
    "thirdPartyStuff": Object {
      "text": "Hello third-party schema!",
    },
  },
}
`;
